packages/asset-swapper/contracts/src/KyberDmmSampler.sol

```diff
@@ -140,17 +153,25 @@ contract KyberDmmSampler
         view
         returns (address[] memory pools)
     {
-        pools = new address[](path.length - 1);
         IKyberDmmFactory factory = IKyberDmmFactory(IKyberDmmRouter(router).factory());
+        pools = new address[](path.length - 1);
         for (uint256 i = 0; i < pools.length; i++) {
-            // Currently only supporting the first pool found at the index
+            // find the best pool
+            address[] memory allPools;
             try
-                factory.getPoolAtIndex
+                factory.getPools
                     {gas: KYBER_DMM_CALL_GAS}
-                    (path[i], path[i + 1], 0)
-                returns (address pool)
+                    (path[i], path[i + 1])
+                returns (address[] memory allPools)
             {
-                pools[i] = pool;
+                uint256 maxSupply = 0;
+                for (uint256 j = 0; j < allPools.length; j++) {
```

Comment:
What happens if `allPools.length == 0`? We just have an invalid pool (null) at that index? Should we revert?

Reply:
fixed it, added a revert if no pools are found